### ISSUE TITLE
Probabilities do not sum to 1.0! Aborting.

### ISSUE THREAD

**Issue by josephhughes** [stephaneguindon/phyml#114]
Hi Stephane,

I'm getting an error "Probabilities do not sum to 1.0! Aborting." when running --ancestral:

```
phyml -i input.phy -d aa --ancestral --sequential -b 100 -m LG -f e -v e --run_id test 


  ////////////////////////////////////.\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
  \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\.//////////////////////////////////////////

        . Sequence filename:				 input.phy
        . Data type:					 aa
        . Alphabet size:				 20
        . Sequence format:				 sequential
        . Number of data sets:				 1
        . Nb of bootstrapped data sets:			 100
        . Compute approximate likelihood ratio test:	 no
        . Model name:					 LG
        . Proportion of invariable sites:		 estimated
        . Number of subst. rate catgs:			 4
        . Gamma distribution parameter:			 estimated
        . 'Middle' of each rate class:			 mean
        . Amino acid equilibrium frequencies:		 empirical
        . Optimise tree topology:			 yes
        . Starting tree:				 BioNJ
        . Add random input tree:			 no
        . Optimise branch lengths:			 yes
        . Minimum length of an edge:			 1e-08
        . Optimise substitution model parameters:	 yes
        . Run ID:					 test
        . Random seed:					 1573729728
        . Subtree patterns aliasing:			 no
        . Version:					 3.3.20190909
        . Byte alignment:				 32
        . AVX enabled:					 yes
        . SSE enabled:					 yes

  ////////////////////////////////////.\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
  \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\.//////////////////////////////////////////



. 109 patterns found (out of a total of 757 sites). 

. 667 sites without polymorphism (88.11%).


. Computing pairwise distances...

. Building BioNJ tree...

. This analysis requires at least 16 MB of memory space.


. Score of initial tree: -3178.30

. Starting first round of SPRs...

	       7s |   1 | lnL=     -3172.4 | depth=    0/   40 | improvements=   0 | delta_lnL=    0.0/ 1000.0   +

. Second round of optimization...

	       9s |   2 | lnL=     -3173.4 | depth=    1/   36 | improvements=   2 | delta_lnL=    0.0/ 1000.0  

. Third round of optimization...

	      13s |   3 | lnL=     -3172.5 | depth=    1/   20 | improvements=   2 | delta_lnL=    0.0/ 1000.0 | triple moves=   5  
	      18s |   4 | lnL=     -3172.4 | depth=    0/   16 | improvements=   0 | delta_lnL=    0.0/  100.0 | triple moves=   5   +

. Final optimisation steps...


. Log likelihood of the current tree: -3172.258402340897191606928.

. Estimating ancestral sequences...
. Probabilities do not sum to 1.0! Aborting.
. p[ 0]=4.48052E-17
. p[ 1]=4.44816E-18
. p[ 2]=1.94823E-18
. p[ 3]=4.54109E-22
. p[ 4]=5.53546E-18
. p[ 5]=1.11591E-16
. p[ 6]=2.05658E-19
. p[ 7]=9.79381E-20
. p[ 8]=6.904E-19
. p[ 9]=2.79991E-15
. p[10]=1.11869E-14
. p[11]=1.13268E-17
. p[12]=0.924773
. p[13]=1.48494E-16
. p[14]=2.48604E-20
. p[15]=1.69007E-18
. p[16]=3.84091E-16
. p[17]=2.35306E-18
. p[18]=2.16743E-18
. p[19]=1.90699E-16

```

Have I mis-specified something?

Thanks for your advice.

**Comment by stephaneguindon** [comment]
It looks as if there's an issue with the -v e option when used in combination with --ancestral. If you can live without using invariant sites, then please feel free to remove -v e from your command line. I'll look into this issue asap.
Regards,
-Stephane-

**Comment by josephhughes** [comment]
Thanks for the quick response. It is running fine without invariant sites. I'd been keen to know when you have managed to find a fix, so I will leave the issue open to be notified when fixed.
Thanks,
Joseph

**Comment by stephaneguindon** [comment]
Hi Joseph,
I think I've found where this issue was coming from and fixed the bug in the latest commit (7d794b5ad2e3affc8bdc0e6321238392fad0b9a2). I'm now closing this issue but please feel free to re-open it if you notice something wrong.
Best regards,
-Stephane-
